### PR TITLE
Fix exceptions propagation from Signal methods

### PR DESCRIPTION
**Pull request by roxblnfk**
## What was changed

Exceptions from Signal methods now will be propagated and affect Workflow.
Deserialization errors will be ignored. Called Signal method in this case will be skipped.

## Checklist

How was this tested: manually; also added autotests.

> **Note**:
> The fix will be cherry-picked into `master` and `2.7` branches after `2.5.3` release

### CODE REVIEW

**Review by wolfy-j — APPROVED** [review]

File: tests/Functional/Client/FailureTestCase.php [hunk 1]
```diff
@@ -80,4 +83,66 @@ public function testChildWorkflowFailurePropagation()
...
+        $run = $client->startWithSignal($wf, 'failActivity', ['foo']);
+
+        try {
+            sleep(5);
+            $wf->failActivity('foo');
+            $this->fail('Signal must fail');
+        } catch (AssertionFailedError $e) {
+            throw $e;
+        } catch (\Throwable $e) {
+            $this->assertInstanceOf(WorkflowNotFoundException::class, $e);
```
Comment (Sushisource) [inline tests/Functional/Client/FailureTestCase.php:141]: Hmm... is there an `ActivityNotFoundException`? Seemingly this should throw that?

**Review by Sushisource — APPROVED** [review]
Approving since I'm pretty sure this makes sense, but, a couple things to double-check.

File: tests/Functional/Client/FailureTestCase.php (hunk 1 above)
Comment (wolfy-j) [reply tests/Functional/Client/FailureTestCase.php:141]: Not really, this error is when we are trying to send a signal to an already dead workflow.

File: tests/Functional/Client/AwaitTestCase.php [hunk 2]
```diff
@@ -133,7 +133,16 @@ public function testFailSignalSerialization()
...
 
         // breaks the invocation
-        $wait->addValue(['hello'], 123);
+        //
+        // Throws:
+        // Temporal\Exception\Failure\ApplicationFailure
+        // Previous:
+        // Temporal\Exception\InvalidArgumentException:
+        // The passed value of type "array" can not be converted to required type "string" in
+        // src\Internal\Declaration\Dispatcher\AutowiredPayloads.php:34
```
[the remark this answers is not in the document]
Comment (wolfy-j) [reply tests/Functional/Client/AwaitTestCase.php:142]: OK, we can make it less strict. Thx for responding.

File: tests/Functional/Client/FailureTestCase.php (hunk 1 above)
Comment (Sushisource) [reply tests/Functional/Client/FailureTestCase.php:141]: Oooh, I see, I missed that it's already start-with-signal failing the activity earlier. Got it.

File: tests/Functional/Client/AwaitTestCase.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (roxblnfk) [reply tests/Functional/Client/AwaitTestCase.php]: Deserialization errors will be ignored. Called Signal method in this case will be skipped.

File: src/Internal/Workflow/Process/Scope.php [hunk 3]
```diff
@@ -322,16 +324,25 @@ function () use ($cancelID): void {
...
     protected function call(callable $handler, ValuesInterface $values): \Generator
     {
-        $this->makeCurrent();
-        $result = $handler($values);
+        try {
+            $this->makeCurrent();
+            try {
+                $result = $handler($values);
+            } catch (InvalidArgumentException) {
+                // Skip deserialization errors
```
Comment (Sushisource) [inline src/Internal/Workflow/Process/Scope.php:332]: Just want to confirm there aren't any other exception types that could result from bad deserialization that need to be included here.

**Review by Sushisource — APPROVED** [review]

File: src/Internal/Workflow/Process/Scope.php (hunk 3 above)
Comment (roxblnfk) [reply src/Internal/Workflow/Process/Scope.php:367]: As you can see it throws SDK's `InvalidArgumentException` only
https://github.com/temporalio/sdk-php/blob/cf2f921a6ca923f7e6ccffc3d12314c445af236a/src/Internal/Declaration/Dispatcher/AutowiredPayloads.php#L30-L36

**Review by wolfy-j — APPROVED** [review]

**Pull request merged**